File: PlacementCellBackend/DTOs/AlumniStatistics.cs
namespace PlacementCellBackend.DTOs
{
    public class AlumniStatistics
    {
        public int TotalAlumni { get; set; }
        public int UniqueCompanies { get; set; }
        public List<CompanyAlumniCount> CompanyWiseCount { get; set; } = new();
    }

    public class PositionCount
    {
        public string Position { get; set; } = string.Empty;
        public int Count { get; set; }
    }

    public class CompanyAlumniCount
    {
        public string CompanyId { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public int TotalAlumniCount { get; set; }
        public List<PositionCount> PositionWiseCount { get; set; } = new();
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/dtos/AlumniStatistics.java
package com.example.placementicsbackend.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlumniStatistics {

    private int totalAlumni;
    private int uniqueCompanies;
    private List<CompanyAlumniCount> companyWiseCount = new ArrayList<>();
}
